services/qs-spark-service/src/main/java/ai/quantumics/api/service/impl/AwsConnectionServiceImpl.java

```diff
@@ -288,6 +288,18 @@ public BucketFileContent getContent(String bucketName, String file) {
         return bucketFileContent;
     }
 
+    @Override
+    public List<AwsDatasourceResponse> searchConnection(String datasourceName) {
+        List<AwsDatasourceResponse> response = new ArrayList<>();
+        List<AWSDatasource> awsDatasource = awsConnectionRepo.findByActiveAndConnectionNameStartingWithIgnoreCaseOrActiveAndConnectionNameEndingWithIgnoreCase(true, datasourceName,true, datasourceName);
```

Comment:
expecting like %searchConnection%

Reply:
AwsDatasourceResponse can be used directly from repo method instead of AWSDatasource

Reply:
To segregate entity and response.